### DISCUSSION TITLE
How to link already existing devices in LibreNMS and Netbox

### DISCUSSION THREAD

**Discussion by pc86** [bonzo81/netbox-librenms-plugin#18] [General]
Hello
Here is the context 
We are using SNMPv3, so using Netbox à Source of Truth is not the good option from now - at least for LibreNMS.
We are using LibreNMS for years, ans we start to implement Netbox.

I did not found a way to link existing devices in both software. 
Any advice is welcome.
Best regards

**Comment by bonzo81** [comment]
Hi @pc86,

When loading the LibreNMS Sync page, the plugin uses the primary IP of the device (or vm) to search LibreNMS for a device that matches. If it does not find a match, the button to 'Add to LibreNMS' is shown.

If you know that the same device exists in LibreNMS, check that the primary IP in Netbox matches your LibreNMS hostname for the same device.

I hope that helps.

**Reply by richardcyrus** [reply]
Hi @bonzo81,

Is there any way to use the DNS name as a way to link devices? My devices in LibreNMS are added with their DNS names as the primary name.

**Reply by bonzo81** [reply]
Hi @richardcyrus,

This will be available in the next release v0.2.8. As long as the device name in NetBox is the same FQDN name used in LibreNMS you should be good to go.

I'm hoping to get this out in the next week.

**Reply by richardcyrus** [reply]
Thank you for the reply. Looking forward to the release. Maybe also consider for a future release, using the DNS name value from the Primary IP Address object as a potential matching source.

**Reply by bonzo81** [reply]
Thanks for this suggestion. I wasn't aware of that field. I think I can work that into the next release, and make the logic for the  device identifier more robust. Thanks again.

**Comment by pc86** [comment]
Thank you for you fast reply.
The device I tested had a Primary IP address assign to a VLAN.
Therefore, I found a way, I replaced the name of the device in LibreNMS with its primary IP adress and it works :)

Best regards

**Comment by kevintdiep1980** [comment]
Do you have any plans to support Netbox v4.5.2?

**Reply by bonzo81** [reply]
I won't be able to test this until next week.  I can say 'unofficially' that version 4.5 is working but I've not fully tested 4.5.x. if you're in a position to try it safely, on a non production Netbox, and feed back here please do!

**Reply by tvo6** [reply]
Hello,
I've just opened issue #231 before reading this discussion. My netbox install (v4.5.2) is not yet production, so I can run some tests on it if needed.
(and thanks for this plugin)
